src/IsolatedExternalsPlugin.ts

```diff
@@ -0,0 +1,195 @@
+import {
+  Compiler,
+  compilation,
+  ExternalsElement,
+  ExternalsObjectElement
+} from 'webpack';
+import fs from 'fs';
+import path from 'path';
+import { promisify } from 'util';
+import { ConcatSource, Source } from 'webpack-sources';
+import randomstring from 'randomstring';
+
+import Compilation = compilation.Compilation;
+
+export interface IsolatedExternalInfo {
+  name: string;
+  url: string;
+  loaded?: boolean;
+}
+
+export interface IsolatedExternalsElement {
+  [key: string]: IsolatedExternalInfo;
+}
+
+export interface IsolatedExternals {
+  [key: string]: IsolatedExternalsElement;
+}
+
+interface Entrypoint {
+  runtimeChunk: {
+    files: string[];
+  };
+}
+
+const readFile = promisify(fs.readFile);
+
+type IsolatedExternalsConfig = {
+  [key: string]: {
+    [key: string]: { url: string };
+  };
+};
+
+function wrapApp(
+  source: Source | string,
+  externals: IsolatedExternalsElement,
+  appName: string
+): ConcatSource {
+  const externalsList = Object.entries(externals);
+  const varNames = externalsList
+    .map(([, external]) => `var ${external.name} = context.${external.name};`)
+    .join(' ');
+  const wrappedSource = new ConcatSource(
+    `function ${appName}(context){`,
+    varNames,
+    source,
+    `}`
+  );
+  return wrappedSource;
+}
+
+async function addLoadExternals(
+  source: Source | string
+): Promise<ConcatSource> {
+  const externalsLocation = path.resolve(__dirname, 'util', 'loadExternals.js');
```

Comment:
Where does `loadExternals.js` come from?

Reply:
That's correct! 😄